### PR TITLE
fix(build): fix initial build args

### LINKED ISSUE #5: Template broken: initial build results in error
Following the installation steps results in build error. 

```
info: syncing channel updates for 'nightly-2024-05-01-x86_64-apple-darwin'
info: latest update on 2024-05-01, rust version 1.80.0-nightly (f705de596 2024-04-30)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
error: component download failed for rust-docs-x86_64-apple-darwin: could not rename downloaded file from '/Users/xxx/.rustup/downloads/6dd022d80f6491f2e97fb91649749a2dc95b0411d2a4291930a31e58a41e0932.partial' to '/Users/xxx/.rustup/downloads/6dd022d80f6491f2e97fb91649749a2dc95b0411d2a4291930a31e58a41e0932'
```

after deleting the cargo.lock file the build results in type error and change is needed to the main.rs _args type. 
from CStr to *mut u8

```
fn main(_args: *mut u8) -> i32 {
    println!("Hello, Rust!");
    0
}
```

### PR DESCRIPTION
**Pull request by MassivDash**
Fix template build

----
- _args *mut u8 type change
-  remove cargo.lock for not locking the nightly versions builds 
- add Cargo.lock to gitignore

### CODE REVIEW

**Comment by MassivDash** [comment]
Fix for issue https://github.com/flipperzero-rs/flipperzero-template/issues/5

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Generated by Cargo
 # will have compiled files and executables
 /target
+cargo.lock
```
Comment (JarvisCraft) [inline .gitignore]:
```suggestion
Cargo.lock
```

**Review by JarvisCraft — CHANGES_REQUESTED** [review]

**Comment by JarvisCraft** [comment]
Thanks for your contribution!

Just a small fix above (since `.gitignore` is case sensitive), but this looks good to me :)

**Review by JarvisCraft — APPROVED** [review]

**Comment by JarvisCraft** [comment]
Thanks for the fix!

**Pull request merged**